internal/auth/auth.go

```diff
@@ -8,11 +8,31 @@ import (
 	"github.com/google/go-containerregistry/pkg/authn/kubernetes"
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/types"
+	k8s "k8s.io/client-go/kubernetes"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 //go:generate mockgen -source=auth.go -package=auth -destination=mock_auth.go
 
+type RegistryAuthGetterFactory interface {
+	NewRegistryAuthGetter(client client.Client, namespacedName types.NamespacedName) RegistryAuthGetter
+	NewServiceAccountRegistryAuthGetter(coreClientSet k8s.Interface, namespace, serviceAccountName string) RegistryAuthGetter
+}
+
+type registryAuthGetterFactory struct{}
+
+func NewRegistryAuthGetterFactory() RegistryAuthGetterFactory {
+	return registryAuthGetterFactory{}
+}
+
+func (registryAuthGetterFactory) NewRegistryAuthGetter(client client.Client, namespacedName types.NamespacedName) RegistryAuthGetter {
+	return NewRegistryAuthGetter(client, namespacedName)
```

Comment:
Why do we need NewRegistryAuthGetter and NewServiceAccountAuthGetter functions? why not just use their implementation in the member functions? I think that after this commit all the code will simply start using RegisterAuthGetterFactory, no?

Reply:
i understand the need for Factory, i mean that Factory New... functions are calling additional functions ( not member functions of Factory). Why do we need this 2 layer call?